preflight_check/check.py

```diff
@@ -0,0 +1,254 @@
+import re
+from . import utils
+from . import task
+from crmsh import utils as crmshutils
+from crmsh import bootstrap as crmshboot
+
+
+def check(context):
+    """
+    Check environment and cluster state if related options enabled
+    """
+    if context.env_check:
+        check_environment()
+    if context.cluster_check:
+        check_cluster()
+    print()
+
+
+def check_environment():
+    """
+    A set of functions to check environment
+    """
+    print("\n============ Checking environment ============")
+    check_my_hostname_resolves()
+    check_time_service()
+    check_watchdog()
+    check_firewall()
+
+
+def check_my_hostname_resolves():
+    """
+    Check hostname resolvable
+    """
+    task_inst = task.TaskCheck("Checking hostname resolvable")
+    with task_inst.run():
+        if not crmshboot.my_hostname_resolves():
+            task_inst.error('''Hostname "{}" is unresolvable.
+  Please add an entry to /etc/hosts or configure DNS.'''.format(utils.this_node()))
+
+
+def check_time_service():
+    """
+    Check time service
+    """
+    task_inst = task.TaskCheck("Checking time service")
+    with task_inst.run():
+        timekeepers = ('chronyd.service', 'ntp.service', 'ntpd.service')
+        timekeeper = None
+        for tk in timekeepers:
+            if crmshutils.service_is_available(tk):
+                timekeeper = tk
+                break
+
+        if timekeeper is None:
+            task_inst.warn("No NTP service found.")
+        else:
+            task_inst.info("{} is available".format(timekeeper))
+            if crmshutils.service_is_enabled(timekeeper):
+                task_inst.info("{} is enabled".format(timekeeper))
+            else:
+                task_inst.warn("{} is disabled".format(timekeeper))
+
+            if crmshutils.service_is_active(timekeeper):
+                task_inst.info("{} is active".format(timekeeper))
+            else:
+                task_inst.warn("{} is not active".format(timekeeper))
+
+
+def check_watchdog():
+    """
+    Verify watchdog device
+    """
+    task_inst = task.TaskCheck("Checking watchdog")
+    with task_inst.run():
+        if not crmshutils.service_is_active("sbd.service") and not crmshboot.check_watchdog():
```

Comment:
Maybe only check service_is_enabled is better, since `sbd` is not a must for many cases.
Also don't need to check the condition of `sbd.service` active, since it will watchdog always exist if "sbd.service" actived

Reply:
Changed!